ultra/docs/agent.md

```diff
@@ -0,0 +1,40 @@
+# ULTRA Agents
+
+An ULTRA agent specification inherits from SMARTS's `AgentSpec`. The baseline agents of ULTRA use the `BaslineAgentSpec` available in `ultra/baselines/agent_spec.py`. However, the following are the attributes of an `AgentSpec` available for customization:  
+* agent interface
+* agent parameters
+* agent policy class
+* observation adapter
+* reward adapter
+* info adapter
+* action adapter
+
+```python
+# The baseline agent specification.
+spec = AgentSpec(
+    interface=AgentInterface(
+        waypoints=Waypoints(lookahead=20),
+        neighborhood_vehicles=NeighborhoodVehicles(200),
+        action=action_type,
+        rgb=False,
+        max_episode_steps=max_episode_steps,
+        debug=True,
+    ),
+    agent_params=dict(
+        policy_params=policy_params, checkpoint_dir=checkpoint_dir
+    ),
+    agent_builder=policy_class,
+    observation_adapter=adapter.observation_adapter,
+    reward_adapter=adapter.reward_adapter,
+    # action_adapter=... (not specified for baseline agents),
+    # info_adapter=... (not specified for baseline agents),
+)
+```
+
+- The `interface` is a SMARTS `AgentInterface` that specifies how the agent will interface with the environment. For example, the types of observations (whether they include images, neighbouring social vehicles, accelerometer data, etc.), vehicle type, and action type can be specified with the agent interface.
+- The `agent_builder` is a class inheriting from SMARTS's `Agent`. This class must define `act` and `step` methods.
+- The `agent_params` are the parameters that are passed to the agent builder class.
```

Comment:
`agent_params` and `agent_builder` should be switched to match the argument list in AgentSpec. Also, maybe we can remove "The" that preceeds each bullet point here.
```suggestion
- The `agent_params` are the parameters that are passed to the agent builder class.
- The `agent_builder` is a class inheriting from SMARTS's `Agent`. This class must define `act` and `step` methods.
```

Reply:
Ahh okay yes for sure, makes sense. I will change this, thanks!